### DISCUSSION TITLE
no problem in windows. but crash in linux and macos. Problem is insert bom signature code line.

### DISCUSSION THREAD

**Discussion by mumin16** [nlohmann/json#4086] [General]
void ExportSymbols(std::vector<std::string> &symbols)
{

  auto CustomSymbols_json = json::parse(R"({"CustomSymbols":[{"ConfigSymbols":[]}]})");
  for (auto &&symbol : symbols)
    CustomSymbols_json["CustomSymbols"].at(0)["ConfigSymbols"].push_back({{"Path", std::string{"Custom\\" + symbol}}, {"Symbol", symbol}});

  std::wofstream fout("./CustomSymbols.json"); // wide output stream
  fout.imbue(std::locale(
      fout.getloc(),
      new std::codecvt_utf16<wchar_t, 0x10ffff, std::little_endian>));

  unsigned char bom[] = {0xFF, 0xFE};
  fout.write((wchar_t *)bom, sizeof(bom) / 2);

  fout << CustomSymbols_json.dump().c_str();
}

**Comment by nlohmann** [comment]
Do you have more information on the crash?
